### ISSUE TITLE
How do I download files from here?

### ISSUE THREAD

**Issue by OptimusMetallus** [BSData/wh40k-7th-edition#365]
https://github.com/BSData/wh40k

How do I get files from there for my Battlescribe Pro iPad app?  I've been downloading files from here...

http://battlescribedata.appspot.com

...but apparently those are out-of-date, and don't include the Knight-Castigator, which I really need. I'd really like to get my hands on up-to-date files, and the first link looks like it might be the place to get them, but for the life of me I can't figure out how you download them from there. Could someone please help me out? Thanks.

**Comment by amis92** [comment]
Hello there.

I have bad news for you. What you're trying to do is absolutely impossible. I'll quote myself from #313 :

> Here on GitHub we communicate, we have the main revision of files, their history and an issue system. We work here. The fruits of our work are then released and a web application hosted by us on AppSpot downloads this new revision. Then, all the other people (not maintainers) can download them, and the update system is triggered in BattleScribe.
> 
> So the appspot page and service is a frontend for our backend here on GitHub.

I hope it's clearer for you now.

Currently there is no way to have GitHub repository in it's _master_ state on iPad or anything other than PC/Mac. You are, unfortunately, forced to wait until we do the release, and release the _master_ to public in it's state for that moment. Then again, we fix up things, change them up, and you wait again, until we do next release.

The thing which happened with Knight Castigator disappearing... that was a mess here. Our fault. Won't possibly happen again. :D

**Comment by Kangodo** [comment]
Can't you link the iPad-app to Dropbox?
I've set up GitHub to place them all in Dropbox and I use that same folder on my Android-phone, so I will always use the Master-branch to create rosters on my phone.

As to the original question: I would recommend using the 'public' files, it usually doesn't take very long before we push our changes to the public.

**Comment by amis92** [comment]
Haha, @Kangodo , that's right, you can do that! :D

Sometimes too long, I think. We should do releases each two days at least, if there are new updates.

**Comment by Kangodo** [comment]
We had the last release 16 hours ago, so is it okay to close this?

Obtaining the files on the phone is amazing, since it allows me to check how stuff looks on Android before committing it.

**Comment by OptimusMetallus** [comment]
Well, I got the update on Battlescribe Pro on my iPad, but everytime I start making an army list it immediately freezes up and I can't do anything. The screen becomes totally unresponsive no matter what button I try touching or what I try doing. I invariably end up having to delete Battlescribe Pro from the iPad entirely and having to redownload the app.

**Comment by OptimusMetallus** [comment]
And I tried uploading those files to Dropbox and copying them to the app, but it didn't work. Using that datafiles page, as far as I know, is my only option to get datafiles for the app.

**Comment by ghost** [comment]
I heard iOS 8 is causing issues is that what ur running?

**Comment by ghost** [comment]
I just have the regular version of bs on my iPhone 5s iOS 7.1.2 and am having no issues

**Comment by OptimusMetallus** [comment]
iOS 8, I think. I downloaded the app to my iPhone 4 and it works fine there, but that goes back to the file not being up-to-date. Like I mentioned, when I first got Battlescribe it had the Cerastus Knight-Castigator in the Imperial Knight list. I downloaded the most recent update to my iPhone and checked it out, but the Cerastus Knight-Castigator is still missing from there, and the Questoris Knight Magaera hasn't been added either.
